src/sync/core-sync-state.js

```diff
@@ -182,13 +182,23 @@ export class CoreSyncState {
    * blocks/ranges that are added here
    *
    * @param {PeerId} peerId
-   * @param {Array<{ start: number, length: number }>} ranges
+   * @param {number} start
+   * @param {number} length
+   * @returns {void}
    */
-  setPeerWants(peerId, ranges) {
+  addWantRange(peerId, start, length) {
```

Comment:
I was worried about performance (because of the cost of calculating state), but I _think_ the way this is implemented, the throttle on SyncState should avoid state calculation happening too frequently, so I think it is ok to just call this with every range (there will be a lot of calls to this).

Reply:
I wrote a [simple benchmark script with Deno](https://gist.github.com/EvanHahn/c03d3b3cb88a6dc708782b83f52a6be1). On my machine, I could call `setRange` 7,364 times per second, at least with my simple benchmark. I realize my computer is probably much faster than most phones, but even a 100x slowdown is probably okay here.

I'm going to leave this as is, but let me know if that's wrong.

Reply:
Sorry for the lack of clarity, I was thinking out loud. I was concerned about the performance not because of the cost of this function, but because it triggers a state update, and reading state is expensive (when cores get large, which they will do for the blob core). However, the way we currently trigger state updates, we don't actually calculate state as part of the `update` event, and then in SyncState we throttle the handler for `update` events so that `getState()` is only called ever 200ms, so with the current implementation this is all fine. The reason this was originally `ranges` was to avoid triggering a state-recalculation for every range set, but our throttle in SyncState solves that problem for now. We just need to remember this in the future.